### ISSUE TITLE
Storage account access tier 'cool'

### ISSUE THREAD

**Issue by TheRSP** [CompositionalIT/farmer#454]
Is there any way to set the access tier on a storage account object? My understanding is that there is a significant cost difference between coold and hot tier access so would like to configure the tier to reduce cost.

```fsharp
  let storageAcc = storageAccount {
    name "myStorageAccount" 
    sku Storage.Sku.Standard_LRS
    
    use_static_website contentPath "index.html"
  }
```
produces
```json
    {
      "apiVersion": "2019-06-01",
      "dependsOn": [],
      "kind": "StorageV2",
      "location": "uksouth",
      "name": "myStorageAccount",
      "properties": {},
      "sku": {
        "name": "Standard_LRS"
      },
      "type": "Microsoft.Storage/storageAccounts"
    }
```
I believe the difference I would like is
```diff
      "location": "uksouth",
      "name": "myStorageAccount",
-     "properties": {},
+     "properties": {
+        "accessTier": "Cool"
+     },
      "type": "Microsoft.Storage/storageAccounts"
    }
```

**Comment by isaacabraham** [comment]
This only applies (AFAIK) for BlobStorage accounts. Taken from the [official docs](https://docs.microsoft.com/en-us/azure/templates/microsoft.storage/storageaccounts):

> Required for storage accounts where kind = BlobStorage. The access tier used for billing. - Hot or Cool

For StorageV2 accounts, individual blobs can be set to hot, cool or archive (and you can do this through e.g. the SDKs). You can also set up access policies to automatically e.g. move from Hot to Cool after some arbitrary time after writing. Policies are [fully supported in Farmer](https://compositionalit.github.io/farmer/api-overview/resources/storage-account/#builder-keywords).

**Comment by TheRSP** [comment]
That line of documentation confused me a bit - it's clear for BlobStorage but doesn't say anything about other kinds.
I tried the above change (taking a local copy of the storageAccount builder, config and ARM model) and it does seem to change the expected setting and changes the access tier on the '$web' container from 'Hot (Inferred)' to 'Cool (Inferred)'.

![image](https://user-images.githubusercontent.com/4334224/98547116-988d5900-228f-11eb-949e-f7ccf1c17e02.png)

Is there a better way to achieve the goal of having all blob containers default to cool tier?

context: we're trying to move to using farmer as the basis for deploying all azure infrastructure within our company and we have a standing poilicy that all storage accounts should be cool tier by default which is why I'd like to have a simple way to set the default access tier.

**Comment by isaacabraham** [comment]
Interesting. Saying $web - are you using this for static web apps?

**Comment by TheRSP** [comment]
Yup. Using it for static website (actual template also has a CDN hence my issue last week which is being fixed)

**Comment by isaacabraham** [comment]
So this is indeed the "default" tier for new blobs. Interesting. I guess it only works for StorageV2 or Blob Storage accounts.

BTW the CDN enhancements are now published as part of 1.3.0-beta1.

**Comment by TheRSP** [comment]
Do you think that adding a `default_access_tier` operator to the storage builder would be a good solution?
